lib/spack/llnl/util/tty/log.py

```diff
@@ -360,34 +360,30 @@ def __init__(self, fd):
 
     @property
     def fd(self):
+        """It is assumed this is accessed for the purpose of fdopen-ing
+        it.
+
+        In that case, GC will attempt to close() both the resulting
+        File and the Connection (if this stores its handle in a
+        Connection); to avoid a double-close issue, nullify the
+        Connection object.
+        """
```

Comment:
This makes no sense to me... you're rebinding `self._connection._handle` and `self._connection` to None. The object are still GC'd...

Reply:
Oh, you meant that unsetting `self._connection._handle` makes it so that `self._connection.__del__` does not call `os.close(self._connection._handle)`.

I may be reading this wrong, but it looks like since you've removed the close call on read_multiprocess_fd, it's now never closed instead of twice.

Why not leave this like it was and let `__del__` close?

Reply:
It's closed at https://github.com/spack/spack/pull/47035/files#r1806572785